### ISSUE TITLE
fix roundcube ssl xtrange errors, but its perfectly working

### ISSUE THREAD

**Issue by mckaygerhard** [hypery2k/owncloud#380]
after commits for improve and compat to newer versions with older working got some xtrange erros, but still working in production 7.X server:

```
Error	roundcube	OC_RoundCube_App.class.php->login(): lenz_gerardo: RoundCube can't login to roundcube due to a login error to roundcube	2017-06-09T12:13:56+00:00
Error	PHP	Undefined index: password at /home/intranet/elfichero/apps/roundcube/lib/RoundCubeAuthHelper.class.php#62
```

need to improve and integrate lasted proposal changes to rouncube tests, **but in production are working perfectly in my RC (1.0.5) + OC (7.X) installation** until the commit c541c23090a636333ec2ea11b96ec572de9e36a4

```
There were 4 errors:

1) OC_RoundCube_App_Test::testKeyGeneration
Undefined variable: config

/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:162
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/src/test/php/OC_RoundCube_App_Test.php:85
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/snippet.php:3

2) OC_RoundCube_App_Test::testCrypt
Undefined variable: config

/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:162
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/src/test/php/OC_RoundCube_App_Test.php:104
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/snippet.php:3

3) OC_RoundCube_App_Test::testSaveManualLoginDataWithErrors
Undefined variable: config

/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:162
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/src/test/php/OC_RoundCube_App_Test.php:124
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/snippet.php:3

4) OC_RoundCube_App_Test::testSaveManualLoginDataWithCryptErrors
Undefined index: publicSSLKey

/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/test-classes/mocks_ocp.php:43
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:196
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:284
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/lib/RoundCubeApp.class.php:554
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/src/test/php/OC_RoundCube_App_Test.php:134
/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/snippet.php:3
```

**Comment by mckaygerhard** [comment]
the changed code was function generateKeyPair(string,string) and the respective test fail.. this are realted to changes in #377

**Comment by mckaygerhard** [comment]
now new errors xtrange, after commits for improve and compat to newer versions with older working got some xtrange erros, but still working in production 7.X server:

```
Error	roundcube	OC_RoundCube_App.class.php->login(): lenz_gerardo: RoundCube can't login to roundcube due to a login error to roundcube	2017-06-09T12:13:56+00:00
Error	PHP	Undefined index: password at /home/intranet/elfichero/apps/roundcube/lib/RoundCubeAuthHelper.class.php#62
```

tests now reports new errors, i dont understand the OX api its a stupid document:

```
  EXCEPTION [org.phpmaven.phpexec.library.PhpErrorException 
Failed to execute PHP with arguments '"/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/snippet.php" --log-junit "/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/phpunit-reports/OC_RoundCube_Login_Test.xml"  "/var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/src/test/php/OC_RoundCube_Login_Test.php"' [Return: 255]:
PHP Fatal error:  Uncaught Error: Call to undefined method OCP\Util::getVersion() in /var/lib/jenkins/jobs/owncloud/jobs/OwnCloud_smoke_tests/workspace/roundcube/target/classes/appinfo/app.php:24
```

trying some fixeds at my local copy and then commit to see what happened..

**Comment by mckaygerhard** [comment]
closes via commit 79fa1a9e05729bc24e1902177429eae173a830d9 now jenkis seems test result are correct.
